dashpay/src/main/kotlin/org/dashevo/dashpay/BlockchainIdentity.kt

```diff
@@ -0,0 +1,767 @@
+/**
+ * Copyright (c) 2020-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package org.dashevo.dashpay
+
+import com.google.common.base.Preconditions
+import com.google.common.collect.ImmutableList
+import com.google.common.util.concurrent.MoreExecutors
+import kotlinx.coroutines.delay
+import org.bitcoinj.core.*
+import org.bitcoinj.crypto.ChildNumber
+import org.bitcoinj.crypto.KeyCrypter
+import org.bitcoinj.evolution.CreditFundingTransaction
+import org.bitcoinj.wallet.DerivationPathFactory
+import org.bitcoinj.wallet.SendRequest
+import org.bitcoinj.wallet.Wallet
+import org.dashevo.dashpay.callback.RegisterIdentityCallback
+import org.dashevo.dashpay.callback.RegisterNameCallback
+import org.dashevo.dashpay.callback.RegisterPreorderCallback
+import org.dashevo.platform.Platform
+import org.dashevo.dapiclient.model.DocumentQuery
+import org.dashevo.dpp.document.Document
+import org.dashevo.dpp.document.DocumentsStateTransition
+import org.dashevo.dpp.identity.Identity
+import org.dashevo.dpp.identity.IdentityPublicKey
+import org.dashevo.dpp.statetransition.StateTransition
+import org.dashevo.dpp.toBase64
+import org.dashevo.dpp.toHexString
+import org.dashevo.dpp.util.Cbor
+import org.dashevo.dpp.util.Entropy
+import org.slf4j.LoggerFactory
+import java.util.*
+import kotlin.collections.ArrayList
+import kotlin.collections.HashMap
+import kotlin.collections.HashSet
+import kotlin.concurrent.timerTask
+
+class BlockchainIdentity {
+
+    var platform: Platform
+    var params: NetworkParameters
+
+    private constructor(params: NetworkParameters) {
+        this.params = params
+        platform = Platform(params)
+    }
+
+    companion object {
+        const val BLOCKCHAIN_USERNAME_SALT = "BLOCKCHAIN_USERNAME_SALT"
+        const val BLOCKCHAIN_USERNAME_STATUS = "BLOCKCHAIN_USERNAME_STATUS"
+
+        private val log = LoggerFactory.getLogger(Peer::class.java)
+    }
+
+    enum class RegistrationStatus {
+        UNKNOWN,
+        REGISTERING,
+        REGISTERED,
+        NOT_REGISTERED
+    }
+
+    enum class UsernameStatus(val value: Int) {
+        NOT_PRESENT(0),
+        INITIAL(1),
```

Comment:
Why `INITAL` and `PREORDER_REGISTRATION_PENDING` have the same value?

Reply:
That is probably a typo.  Will check.

Reply:
This is fixed here:

https://github.com/dashevo/android-dashpay/pull/1/commits/f50c06278f5503dbf3fab80b0b01daefa199e8be